src/visudo/mod.rs

```diff
@@ -104,6 +114,18 @@ fn check(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()> {
         }
     }
 
+    if metadata.uid() != User::real_uid().inner() {
+        return Err(io::Error::new(
```

Comment:
Maybe we can factor this out into a small function that compares a `uid` with `User::real_uid().inner()` and then gives this error? For the sake of DRY (I now this is only a simple call and we only repeat ourselves once, but `cargo fmt` makes it seem worse than it is..)